sharding/indexHashedNodesCoordinator.go

```diff
@@ -18,6 +24,7 @@ type indexHashedNodesCoordinator struct {
 	shardConsensusGroupSize int
 	metaConsensusGroupSize  int
 	selfPubKey              []byte
+	consensusGroupCache     NodesCoordinatorCache
```

Comment:
why did you defined a new type for cache ?

Reply:
I wanted a decoupling from the Cacher because I just need the Put here Get methods here